File: Python/majority-element/solution2.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        hashmap = {}

        for num in nums:
            if num not in hashmap:
                hashmap[num] = 1
            else:
                hashmap[num] += 1
        
        for num, count in hashmap.items():
            if count > len(nums) / 2:
                return num
        

        return 0

File: Java/longest-consecutive-sequence/Solution2.java
//This also uses hashset and sorting algorithm.

class Solution {
    public int longestConsecutive(int[] nums) {
        int count = 1;
        int result = 1;

        if (nums.length == 0) {
            return 0;
        }

        HashSet<Integer> set = new HashSet<>();

        for (int num : nums) {
            set.add(num);
        }

        ArrayList<Integer> arr_list = new ArrayList<>(set);
        Collections.sort(arr_list);

        for (int i = 0; i < arr_list.size() - 1; i++) {
            if (arr_list.get(i + 1) - arr_list.get(i) == 1) {
                count++;
            } else {
                count = 1;
            }

            result = Math.max(count, result);
        }

        return result;
    }
}
